src/session_pool/session.js

```diff
@@ -172,4 +178,40 @@ export class Session {
         this.errorScore += 1;
         this.usageCount += 1;
     }
+
+    /**
+     * Retires session based on status code.
+     * @param statusCode {Number} - HTTP status code
+     * @return {boolean} whether the session was retired.
+     */
+    checkStatus(statusCode) {
+        const isBlocked = STATUS_CODES_BLOCKED.includes(statusCode);
+        if (isBlocked) {
+            this.retire();
+        }
+        return isBlocked;
+    }
+
+    /**
+     * Sets cookies from response to the cookieJar.
+     * Parses cookies from `set-cookie` header and sets them to `Session.cookieJar`.
+     * @param response
+     */
+    setCookiesToJar(response) {
+        const cookies = getCookiesFromResponse(response).filter(c => c);
+
+        for (const cookie of cookies) {
+            this.cookieJar.setCookieSync(cookie, response.url, { ignoreError: true });
```

Comment:
We should use the `async` methods here to prevent blocking the event loop.

Reply:
When we use in-memory cookie storage there is no need for async. The async version should be used for asynchronous stores. https://www.npmjs.com/package/tough-cookie#setcookiesynccookieorstring-currenturl-options